kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/ProcessCodegen.java

```
@@ -582,4 +595,50 @@ public Optional<ApplicationSection> section() {
         return 10;
     }
+
+    private void processProducerTemplate(Optional<Expression> statement) {
+        TemplatedGenerator generator = TemplatedGenerator.builder()
+                .withTemplateBasePath("/class-templates/producer/")
+                .withFallbackContext(JavaKogitoBuildContext.CONTEXT_NAME)
+                .withTargetTypeName(BUSINESS_CALENDAR_PRODUCER_TEMPLATE)
+                .build(context(), BUSINESS_CALENDAR_PRODUCER_TEMPLATE);
+        CompilationUnit compilationUnit = generator.compilationUnitOrThrow();
+
+        statement.ifPresent(expression -> compilationUnit.findFirst(ClassOrInterfaceDeclaration.class)
+                .flatMap(NodeWithMembers::getDefaultConstructor)
+                .ifPresent(constructor -> constructor.getBody().getStatements().stream()
+                        .filter(Statement::isExpressionStmt)
+                        .flatMap(stmt -> stmt.getChildNodes().stream())
+                        .filter(AssignExpr.class::isInstance)
+                        .map(AssignExpr.class::cast)
+                        .filter(assignExpr -> assignExpr.getTarget() instanceof FieldAccessExpr field &&
+                                BUSINESS_CALENDAR_FIELD_NAME.equals(field.getNameAsString()))
+                        .forEach(assignExpr -> assignExpr.setValue(expression))));
+
+        storeFile(PRODUCER_TYPE, generator.generatedFilePath(), compilationUnit.toString());
+    }
+
+    private Optional<Expression> deriveBusinessCalendarDeclarationStatement(String businessCalendarClassName) {
+        Class<?> businessCalendarClass;
+        boolean validBusinessCalendar = false;
+        ObjectCreationExpr declarationStmt = null;
+        if (Objects.isNull(businessCalendarClassName)) {
+            return Optional.empty();
+        }
+        try {
+            businessCalendarClass = Class.forName(businessCalendarClassName, true, context().getClassLoader());
+        } catch (ClassNotFoundException e) {
```

As explained time ago, we can't simply swallow up errors and exceptions: in that case, there is some configuration error of which the user must be aware, somehow: the application build should fail here with a clear message